pkg/apis/hive/v1alpha1/validating-webhooks/syncset_validating_admission_hook.go

```diff
@@ -233,14 +246,60 @@ func (a *SyncSetValidatingAdmissionHook) validateUpdate(admissionSpec *admission
 	}
 }
 
-func checkValidPatchTypes(patches []hivev1.SyncObjectPatch) (string, bool) {
+func validatePatches(patches []hivev1.SyncObjectPatch, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
 	validTypes := map[string]struct{}{"json": {}, "merge": {}, "strategic": {}}
 	for _, patch := range patches {
 		if _, ok := validTypes[patch.PatchType]; !ok {
-			return patch.PatchType, false
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("PatchType"), patch.PatchType,
+				"Invalid patch type found: %s. Valid patch types are: json, merge, and strategic"))
+		}
+	}
+	return allErrs
+}
+
+func validateResources(resources []runtime.RawExtension, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, resource := range resources {
+		allErrs = append(allErrs, validateResource(resource, fldPath.Index(i))...)
+	}
+	return allErrs
+}
+
+func validateResource(resource runtime.RawExtension, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	rType := &metav1.TypeMeta{}
+	err := json.Unmarshal(resource.Raw, rType)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, "resource", "Unable to unmarshal resource Kind and APIVersion"))
+		return allErrs
+	}
+
+	invalidResourceGroupKinds := map[string][]string{
+		"authorization.openshift.io": {
+			"Role",
+			"RoleBinding",
+			"ClusterRole",
+			"ClusterRoleBinding",
+			"SubjectAccessReview",
+		},
+	}
+
+	for invalidGroup, kinds := range invalidResourceGroupKinds {
```

Comment:
If you make `invalidResourceGroupKinds` a `map[string]map[string]bool`, then you can check if a given group/kind is invalid more concisely with the following.
```
invalidResourceGroupKinds[rType.GroupVersionKind().Group][rType.Kind]
```